browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -356,5 +367,100 @@ void BraveTabContainer::PaintChildren(const views::PaintInfo& paint_info) {
   }
 }
 
+BrowserRootView::DropIndex BraveTabContainer::GetDropIndex(
+    const ui::DropTargetEvent& event) {
+  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs) ||
+      !tabs::utils::ShouldShowVerticalTabs(
+          tab_slot_controller_->GetBrowser())) {
+    return TabContainerImpl::GetDropIndex(event);
+  }
+
+  // Force animations to stop, otherwise it makes the index calculation tricky.
+  CompleteAnimationAndLayout();
+
+  const int x = GetMirroredXInView(event.x());
+  const int y = event.y();
+
+  std::vector<TabSlotView*> views = layout_helper_->GetTabSlotViews();
+
+  // Loop until we find a tab or group header that intersects |event|'s
+  // location.
+  for (TabSlotView* view : views) {
+    const int max_y = view->y() + view->height();
+    if (y >= max_y) {
+      continue;
+    }
+
+    if (view->GetTabSlotViewType() == TabSlotView::ViewType::kTab) {
+      Tab* const tab = static_cast<Tab*>(view);
+
+      // Closing tabs should be skipped.
+      if (tab->closing()) {
+        continue;
+      }
+
+      if (tab->data().pinned && !IsPointInTab(tab, gfx::Point(x, y))) {
+        continue;
+      }
+
+      // When a tab is dropped between pinned tabs, it will add a new tab
+      // among unpinned tabs. As a result, there's no need to check the hot
+      // height for pinned tabs.
+      const int model_index = GetModelIndexOf(tab).value();
+      if (!tab->data().pinned) {
+        const bool first_in_group =
+            tab->group().has_value() &&
+            model_index ==
+                controller_->GetFirstTabInGroup(tab->group().value());
+
+        // When hovering over the top or bottom quarter of a tab, the drop
+        // indicator will point between tabs.
+        const int hot_height = tab->height() / 4;
+
+        if (y >= (max_y - hot_height)) {
+          return {model_index + 1, true /* drop_before */,
+                  false /* drop_in_group */};
+        }
+
+        if (y < tab->y() + hot_height) {
+          return {model_index, true /* drop_before */, first_in_group};
+        }
+      }
+
+      return {model_index, false /* drop_before */, false /* drop_in_group */};
+
```

Comment:
nit: unnecessary empty line
```suggestion
      return {model_index, false /* drop_before */, false /* drop_in_group */};
```